Lakebase docs (#7696)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/connect-data-platform/connect-redshift-postgresql-alloydb.md

---
title: "Connect Redshift, PostgreSQL, Lakebase and AlloyDB"
id: connect-redshift-postgresql-alloydb

> description: "Setup instructions for connecting Redshift, PostgreSQL,Lakebase and AlloyDB to dbt"

Reviewer: ```suggestion
description: "Setup instructions for connecting Redshift, PostgreSQL, Lakebase, and AlloyDB to dbt"
```

> sidebar_label: "Connect Redshift, PostgreSQL, Lakebase and AlloyDB"

Reviewer: ```suggestion
sidebar_label: "Connect Redshift, PostgreSQL, Lakebase, and AlloyDB"
```

---

> /* TODO: break Redshift connection docs to separate page */ 

Reviewer: ```suggestion
<!-- TODO: break Redshift connection docs to separate page -->
```

Author: omg I didn't push my changes. thank you!

> dbt Platform supports connecting to PostgresSQL, Postgres compatible databases (AlloyDB, Lakebase) and Redshift. 

Reviewer: ```suggestion
dbt Platform supports connecting to PostgresSQL, Postgres-compatible databases (AlloyDB, Lakebase), and Redshift. 
```

The following fields are required when creating a connection:

| Field | Description | Examples |
| ----- | ----------- | -------- |

> | Host Name | The hostname of the Postgres, Redshift, Lakebase or AlloyDB database to connect to. This can either be a hostname or an IP address. | `xxx.us-east-1.amazonaws.com` or `hostname.us-east-1.redshift.amazonaws.com` or `workgroup-name.123456789.us-east-1.redshift-serverless.amazonaws.com` |

Reviewer: @amychen1776 let's move the databases to the examples, but can you verify that I matched the examples to databases correctly?
```suggestion
| Host Name | The hostname of the database to connect to. This can either be a hostname or an IP address. Refer to [set up pages](/docs/core/connect-data-platform/about-core-connections) to find the hostname for your adapter. | Postgres: `xxx.us-east-1.amazonaws.com` |
```

Author: @runleonarun Lakebase and Alloydb are wrong but I feel like this section is so overcrowded. Any opinions on how we could just trim it down to one and just point people to the right place for their platform specific? We call out what the database hostname is on the Lakebase setup page

Reviewer: Hey @amychen1776 
>Any opinions on how we could just trim it down to one and just point people to the right place for their platform specific? 

I'm aligned that it's too crowded. I don't see hostname available for AlloyDB in their config pages. We could just say something like refer to the config page if we add the hostname info there. Where can I find the hostname infor?

Author: We provide examples as it makes sense in the adapter's specific set up guides for core. I think it's okay to include the aws hostname as an example and then put a note to check out the configs?

| Port | Usually 5432 (Postgres) or 5439 (Redshift) | `5439` |
| Database | The logical database to connect to and run queries against. | `analytics` |

**Note**: When you set up a Redshift or Postgres connection in <Constant name="cloud" />, SSL-related parameters aren't available as inputs. 

<Lightbox src="/img/docs/dbt-cloud/cloud-configuring-dbt-cloud/postgres-redshift-connection.png" width="70%" title="Configuring a Redshift connection"/>

### Authentication Parameters

For authentication, <Constant name="cloud" /> users can use **Database username and password** for Postgres and Postgres compatible databases. For more information on what is supported, check out the database specific setup page for limitations and helpful tips. 

> In addition, for Redshift, users can use **IAM User authentication** via [extended attributes](/docs/dbt-cloud-environments#extended-attributes) or Identity Center via [external Oauth](/docs/cloud/manage-access/external-oauth) 

Reviewer: @amychen1776 I thought Redshift was limited to username/password for Fusion. Should we mention that? See [this conversation](https://dbt-labs.slack.com/archives/C0897L5FJ4U/p1754071422964219?thread_ts=1754069025.780879&cid=C0897L5FJ4U).

Author: Yes - for Fusion that is correct. @runleonarun I would like to propose that we break out Redshift from this page to its own (it's different enough at this point that this is just too crowded and no one should be using the dbt-postgres adapter to link to Redshift) and then add in the Fusion callouts? Could someone on the docs team help with that and I can review the PR? I think this should be a priority because of the Fusion components so it's more clear

Reviewer: OK I will see what we can do. First I'll work on getting this one merged!

Author: thank you!
